pkg/cmd/step/helm/step_helm_apply.go

```diff
@@ -546,3 +500,75 @@ func (o *StepHelmApplyOptions) fetchSecretFilesFromVault(dir string, store confi
 	}
 	return files, nil
 }
+
+// Sets the Release name
+func setReleaseName(ns string, args []string, options *StepHelmApplyOptions) (releaseName string, err error) {
+
+	helmBinary, noTiller, helmTemplate, err := options.TeamHelmBin()
+	if err != nil {
+		return
+	}
+
+	_, devNs, err := options.KubeClientAndDevNamespace()
+	if err != nil {
+		return
+	}
+
+	if releaseName == "" {
+		if len(args) > 0 {
+			releaseName = args[0]
+		} else
+		if devNs == ns {
+			releaseName = platform.JenkinsXPlatformRelease
+		} else {
+			releaseName = ns
+			if helmBinary != "helm" || noTiller || helmTemplate {
+				releaseName = "jx"
+			}
+		}
+	}
+	return
+}
+
+// Unpack all the dependencies
+func unpackDependencies(dir string, secretURLClient secreturl.Client) (err error) {
+	dependencies, err := filepath.Glob(filepath.Join(dir, "charts", "*.tgz"))
+	if err != nil {
+		return errors.Wrapf(err, "finding chart dependencies in %s", filepath.Join(dir, "charts"))
+	}
+	for _, src := range dependencies {
+		dest, err := ioutil.TempDir("", "")
+		if err != nil {
+			return errors.Wrapf(err, "creating temp dir")
+		}
```

Comment:
The temp dir doesn't seem to be cleaned up. You should call: `os.RemoveAll(dest)` at the end of the for loop.

For clarity, you can move the whole loop into a function closure and call `defer os.RemoveAll(dest)` which will ensure that the temporary directory is removed on every return statement.